src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1593,15 +1593,19 @@ public override bool IsDBNull(int ordinal)
         /// <param name="ordinal">The zero-based column to be retrieved.</param>
         /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
         /// <returns><b>true</b> if the specified column value is equivalent to <see cref="DBNull"/> otherwise <b>false</b>.</returns>
-        public override async Task<bool> IsDBNullAsync(int ordinal, CancellationToken cancellationToken)
+        public override Task<bool> IsDBNullAsync(int ordinal, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
             CheckRowAndGetField(ordinal);
 
             if (!_isSequential)
-                return IsDBNull(ordinal);
+                return IsDBNull(ordinal) ? PGUtil.TrueTask : PGUtil.FalseTask;
 
             using (NoSynchronizationContextScope.Enter())
+                return IsDBNullAsyncInternal();
```

Comment:
Um... How this should work?

Reply:
Do you see an issue here? For sequential readers - where IsDBNullAsync may return asynchronously, we call into the local IsDBNullAsyncInternal method, but wrap it with NoSynchronizationContextScope.

Reply:
Got it. Looks to tricky with usings around async calls.

Reply:
Yeah, it is very tricky. See the original comment for the scenario why this is important...